packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2704,14 +2710,90 @@ class _Location {
   }
 }
 
+/// Describer to gather important information about [offendingContext].
+///
+/// This describer is only applicable when the [offendingContext] is created in
+/// widget layer.
+///
+/// This function will be registered to [FlutterErrorDetails.offendingContextDescribers]
+/// in [WidgetsBinding.initInstances].
+Iterable<DiagnosticsNode> describeOffendingContext(dynamic offendingContext) {
+  Element element;
+  try {
+    element = offendingContext is Element ? offendingContext : offendingContext.element;
+  } on NoSuchMethodError {
+    return <DiagnosticsNode>[];
+  }
+  return <DiagnosticsNode>[
+    _describeRelevantUserCode(element),
+  ].where((DiagnosticsNode node) => node != null);
+}
+
+DiagnosticsNode _describeRelevantUserCode(Element element) {
+  if (!WidgetInspectorService.instance.isWidgetCreationTracked()) {
+    return ErrorDescription(
+      'Widget creation tracking is currently disabled. Enabling'
+      ' it enables improved error messages. It can be enabled by passing '
+      '`--track-widget-creation` to `flutter run` or `flutter test`.',
```

Comment:
nit: this string literal is inconsistent about whether spaces should go at the end of the line or the start of the line. We generally prefer the end of the line.